src/store/sagas/visualization-saga.js

```diff
@@ -8,7 +8,20 @@ function* loadVisualizationSaga() {
 }
 
 function* loadVisualization({ value: id }) {
-  yield callEndpoint(() => AuthenticatedHTTPClient.get(`/api/v1/visualization/${id}`), visualizationLoadSuccess, visualizationLoadFailure)
+  try {
+    const response = yield call(AuthenticatedHTTPClient.get, `/api/v1/visualization/${id}`)
+
+    if (response.status < 400) {
+      yield put(setChartInformation(response.data.chart))
+      yield put(setQueryText(response.data.query))
+      yield put(executeFreestyleQuery(response.data.query))
+      yield put(visualizationLoadSuccess(response.data))
+    } else {
+      yield put(visualizationLoadFailure(response.status))
+    }
+  } catch (e) {
+    yield put(visualizationLoadFailure(e.message))
+  }
```

Comment:
It's no big deal, but would it make sense to extract this to its own function to be consistent with `saveVisualization`?

Reply:
The part of @bbrewer-papa-x that loves symmetry likes this idea.  We think it's a bit premature though.